Keter/Proxy.hs

```diff
@@ -62,28 +62,25 @@ reverseProxy useHeader manager hostLookup listener =
 
 withClient :: Bool -- ^ is secure?
            -> Bool -- ^ use incoming request header for IP address
+           -> Int  -- ^ time bound for connections
            -> Manager
            -> HostLookup
            -> Wai.Application
-withClient isSecure useHeader manager portLookup req0 sendResponse =
-    timeBound (5 * 60 * 1000 * 1000) (waiProxyToSettings getDest def
+withClient isSecure useHeader bound manager portLookup req0 sendResponse =
+    if bound > 0
+       then timeBound (bound * 1000) task
+       else task
+  where
+    task = waiProxyToSettings getDest def
         { wpsSetIpHeader =
             if useHeader
                 then SIHFromHeader
                 else SIHFromSocket
-        } manager req0 sendResponse)
-  where
+        } manager req0 sendResponse
     protocol
         | isSecure = "https"
         | otherwise = "http"
 
-    -- FIXME This is a temporary workaround for
-    -- https://github.com/snoyberg/keter/issues/29. After some research, it
```

Comment:
I'd still like a reference to that issue around the timeBound code, as it's likely to confuse someone in the future.